--- src/lib/disk.c.orig	2013-12-07 10:10:52.000000000 -0800
+++ src/lib/disk.c	2013-12-07 10:11:33.000000000 -0800
@@ -807,7 +807,7 @@
 		*((uint32_t *) (void *)data + 4) =
 				(uint32_t) ISCSI_HTONL((uint32_t) sizeof(key));
 				/* length in bytes of list of keys */
-		*((uint64_t *) (void *)data + 8) = (uint64_t) ISCSI_HTONLL(key);
+		*((uint64_t *) ((void *)data + 8)) = (uint64_t) (ISCSI_HTONLL(key));
 		return 8 + sizeof(key);
 	case PERSISTENT_RESERVE_IN_REPORT_CAPABILITIES:
 		(void) memset(data, 0x0, 8);

